### ISSUE TITLE
Driver and boundary trajectories bug

### ISSUE THREAD

**Issue by FMalerba** [tensorflow/agents#390]
The current implementation of the drivers feeds to the observers the trajectories even when they are boundaries (i.e. first step is the last step of the previous episode, and next step is the first step of the new episode). These trajectories are almost always meaningless I believe and the issue in particular is that amongst the observers there's the replay buffer, meaning that it will get filled even with these bad trajectories. Luckily the agent's loss function (looking at dqn agent) discards these trajectories from the learning process, but I still fail to see why the observers should get to see them in the first place...

**Comment by summer-yue** [comment]
Hi, thanks for raising this! We are aware that many replay buffers can contain boundary conditions that get processed by the agent. The agents would mask out these boundary conditions. Agree that this is a bit complex, but shouldn't be causing any problems.

For our newest version of API that we plan to soft launch in July, by default the replay buffer will contain full episodes without boundary trajectories. Stay tuned! :)
